test/cocotb/osdtestlib/eval_utility.py

```diff
@@ -4,59 +4,87 @@
 from cocotb.result import TestFailure
 from cocotb.triggers import RisingEdge
 
+types = ['REG', 'PLAIN', 'EVENT']
+
+subtypes = ['REQ_READ_REG_16', 'REQ_READ_REG_32', 'REQ_READ_REG_64',
+            'REQ_READ_REG_128', 'RESP_READ_REG_SUCCESS_16',
+            'RESP_READ_REG_SUCCESS_32', 'RESP_READ_REG_SUCCESS_64',
+            'RESP_READ_REG_SUCCESS_128', 'RESP_READ_REG_ERROR',
+            'REQ_WRITE_REG_16', 'REQ_WRITE_REG_32', 'REQ_WRITE_REG_64',
+            'REQ_WRITE_REG_128', 'RESP_WRITE_REG_SUCCESS',
+            'RESP_WRITE_REG_ERROR']
+
 class EvalUtility:
     """
-    Provides an additional method to check if a register which was read out 
-    contains the correct values
+    Provides additional methods to check if a packet which was read contains the
+    correct values
     """
 
     @cocotb.coroutine
-    def evaluate_response(self, dut, response, destination, source, reg_width,
-                          expected_payload):
-        """Compare a received payload with its expectation value and output
+    def evaluate_packet(self, dut, packet, mask, destination, source, type,
```

Comment:
Please consider moving this function into DiPacket, e.g. as `DiPacket.equal_to(self, other_packet)`. Also pass another instance of DiPacket, not another instance.